src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php

```diff
@@ -0,0 +1,231 @@
+			'payment_method_types'   => $payment_methods,
+			'application_fee_amount' => (string) $fee->get_integer(),
+		];
+
+		$args = [
+			'body' => $body,
+		];
+
+		$url = 'payment_intents';
+
+		$payment_intent = Requests::post( $url, $query_args, $args );
+
+		if ( ! isset( $payment_intent['id'] ) && ! empty( $payment_intent['errors'] ) ) {
+			$compiled_errors = static::compile_errors( $payment_intent );
+
+			return new \WP_Error(
+				'tec-tc-stripe-test-payment-intent-failed',
+				__( sprintf( 'Stripe reports that it is unable to process charges with the selected Payment Methods. Usually this means that one of the methods selected is not available or not configured in your Stripe account. The errors you see below were returned from Stripe, please correct any inconsistencies or contact Stripe support, then try again: <div class="stripe-errors">%s</div>', $compiled_errors ), 'event-tickets' )
+			);
+		}
+
+		static::cancel( $payment_intent['id'] );
+
+		return true;
+	}
+
+	/**
+	 * Calls the Stripe API and returns a new PaymentIntent object, used to authenticate
+	 * front-end payment requests.
+	 *
+	 * @since TBD
+	 *
+	 * @param Value $value the value object to create a payment intent for.
+	 * @param bool  $retry is this a retry?
+	 *
+	 * @return mixed
+	 */
+	public static function create( Value $value, $retry = false ) {
+		$fee = Application_Fee::calculate( $value );
+
+		$query_args = [];
+		$body       = [
+			'currency'               => $value->get_currency_code(),
+			'amount'                 => (string) $value->get_integer(),
+			'payment_method_types'   => tribe( Merchant::class )->get_payment_method_types( $retry ),
+			'application_fee_amount' => (string) $fee->get_integer(),
+		];
+
+		$stripe_statement_descriptor = tribe_get_option( Settings::$option_statement_descriptor );
+
+		if ( ! empty( $stripe_statement_descriptor ) ) {
+			$body['statement_descriptor'] = substr( $stripe_statement_descriptor, 0, 22 );
+		}
+
+		$args = [
+			'body' => $body,
+		];
+
+		$url = 'payment_intents';
+
+		return Requests::post( $url, $query_args, $args );
+	}
+
+	public static function create_from_cart( Cart $cart, $retry = false ) {
+		$items = tribe( Order::class )->prepare_cart_items_for_order( $cart );
+		$value = tribe( Order::class )->get_value_total( array_filter( $items ) );
+
+		return static::create( $value, $retry );
+	}
+
+	/**
+	 * Calls the Stripe API and returns an existing Payment Intent based ona PI Client Secret.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $payment_intent_id Payment Intent ID formatted from client `pi_*`
+	 *
+	 * @return array|\WP_Error
+	 */
+	public static function get( $payment_intent_id ) {
+		$query_args = [];
+		$body       = [
+		];
+		$args       = [
+			'body' => $body,
+		];
+
+		$payment_intent_id = urlencode( $payment_intent_id );
+		$url               = '/payment_intents/{payment_intent_id}';
+		$url               = str_replace( '{payment_intent_id}', $payment_intent_id, $url );
```

Comment:
Same here. Seems like `sprintf()` would be better.
```suggestion
		$url               = sprintf( '/payment_intents/%s', $payment_intent_id );
```